model/src/lib.rs

```diff
@@ -159,12 +160,23 @@ impl OrderCreation {
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
 pub struct OrderUid(pub [u8; 56]);
 
+impl Display for OrderUid {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut bytes = [0u8; 56 * 2];
+        // Unwrap because the length is always correct.
+        hex::encode_to_slice(&self.0, &mut bytes).unwrap();
+        // Unwrap because the string is always valid utf8.
+        let str = std::str::from_utf8(&bytes).unwrap();
+        f.write_str(str)
```

Comment:
Why not simply?
```suggestion
        f.write_str(&hex::encode(self.0.iter()))
```

Reply:
That version always allocates a string while encode_to_slice doesn't . Not a big deal but I've been trying to avoid this in several places.